frontend/src/metabase/dashboard/actions/writeback.ts

```diff
@@ -233,44 +235,48 @@ export type ExecuteRowActionPayload = {
   dashboard: Dashboard;
   dashcard: ActionDashboardCard;
   parameters: ParameterMappedForActionExecution[];
-  extra_parameters: ParameterMappedForActionExecution[];
+  extra_parameters: ArbitraryParameterForActionExecution[];
+  dispatch: Dispatch;
+  shouldToast?: boolean;
 };
 
-export const executeRowAction = ({
+export const executeRowAction = async ({
   dashboard,
   dashcard,
   parameters,
   extra_parameters,
-}: ExecuteRowActionPayload) => {
-  return async function (dispatch: any) {
-    try {
-      const result = await ActionsApi.execute({
-        dashboardId: dashboard.id,
-        dashcardId: dashcard.id,
-        parameters,
-        extra_parameters,
-      });
-      if (result["rows-affected"] > 0) {
-        dispatch(reloadDashboardCards());
-        dispatch(
-          addUndo({
-            toastColor: "success",
-            message: t`Successfully executed the action`,
-          }),
-        );
-      } else {
-        dispatch(
-          addUndo({
-            toastColor: "success",
-            message: t`Success! The action returned: ${JSON.stringify(result)}`,
-          }),
-        );
-      }
-    } catch (err) {
-      console.error(err);
-      const message =
-        (<any>err)?.data?.message ||
-        t`Something went wrong while executing the action`;
+  dispatch,
+  shouldToast = true,
```

Comment:
added a `shouldToast` flag so we can turn off the toasts when we have inline forms and we can show error or success states in the form component.